deltachat-ios/AppDelegate.swift

```diff
@@ -61,9 +61,35 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         dcAccounts.logger = DcLogger()
         dcAccounts.openDatabase()
         migrateToDcAccounts()
+
+        if let sharedUserDefaults = UserDefaults.shared, !sharedUserDefaults.bool(forKey: UserDefaults.hasSavedKeyToKeychain) {
+            // we can assume a fresh install -> reset the keychain for the case the app was removed and reinstalled
+            if !KeychainManager.deleteDBSecrets() {
+                logger.warning("Failed to delete DB secrets")
+            }
+        }
+
+        let accountIds = dcAccounts.getAll()
+        for accountId in accountIds {
+            let dcContext = dcAccounts.get(id: accountId)
+            if !dcContext.isOpen() {
+                do {
+                    let secret = try KeychainManager.getAccountSecret(accountID: accountId)
```

Comment:
`getAccountSecret()` creates/overwrites the secret if it is not there - but can that happen? maybe `queryAccountSecret()` is the safer choice here.

alternatively, it is sufficient to make `getAccountSecret()` safer, see other comment.

Reply:
same here: getAccountSecret will never overwrite existing passwords